elfloader-tool/src/arch-riscv/crt0.S

```diff
@@ -153,6 +160,13 @@ secondary_harts:
   la s0, secondary_entry
   jr s0
 #endif
+/* If we get here then the HSM extension exists and the current
```

Comment:
could put `hsm_suspend_hart` into the "#if CONFIG_MAX_NUM_NODES > 1" block, becuase it will only be called on SMP

Reply:
It should still be needed to fall through when `CONFIG_MAX_NUM_NODES == 1` and the boot core is not `CONFIG_FIRST_HART_ID`